paper-typeahead.js

```diff
@@ -249,10 +255,19 @@
           this, results, typedValue, dataKey).slice(0, maxResults);
 
         this.set('filteredItems', filteredItems);
-        this.set('_hideResults', filteredItems.length === 0);
+        this.set('_hideResults', !this._canShowResults(filteredItems));
       });
     },
 
+    /**
+     * @private
+     * @param {!Array<?>} results
+     * @return {boolean}
+     */
+    _canShowResults: function(results) {
+      return document.activeElement === this.$.input && results && results.length > 0;
```

Comment:
To work on shadow dom, this needs something like `document.activeElement === this.$.input || this.root.activeElement === this.$.input` unfortunately :-(